tasks/RevolutionTask.cpp

```diff
@@ -130,11 +155,13 @@ void RevolutionTask::evaluateDriveCommand()
         return;
     }
 
+    base::commands::LinearAngular6DCommand compensated_drive = compensateDriveCommand(drive);
+
     string drive_command =
         m_message_parser.parseDriveRevolutionCommandMessage(m_api_version,
             m_devices_id.revolution,
             m_devices_model.revolution,
-            drive,
+            compensated_drive,
```

Comment:
remove all traces of the buoyancy compensation parameter, otherwise it will affect the already compensated_drive parameter again